tests/test_project/admin.py

```diff
@@ -24,9 +25,32 @@ class OperatorInline(admin.StackedInline):
     extra = 0
 
 
-class ProjectAdmin(admin.ModelAdmin):
+class ProjectAdmin(admin.ModelAdmin, UUIDAdmin):
     inlines = [OperatorInline]
     list_display = ['name']
+    readonly_fields = ['uuid']
+    fields = ['name', 'key', 'uuid']
+
+    class Media:
+        js = ('openwisp-utils/js/uuid.js',)
+
+    def _get_fields(self, fields, request, obj=None):
```

Comment:
why are you adding this code and the lines below?

Reply:
I adding those lines so that the readonly fields (`uuid` in this case) should be should come up only in the change_view. without this, the `uuid` field will appear in the `add_view` with a None value. Equally this was the same approach used in `OrganizationAdmin` found in the `openwisp_users` repo.